File: frontend/src/lib/types/travel-guide.ts
export interface TravelGuideCategory {
  id: number;
  name: string;
  slug: string;
  icon?: string;
  description?: string;
  order_index: number;
  is_active: boolean;
  created_at?: string;
  updated_at?: string;
}

export interface TravelGuideCategoryCreate {
  name: string;
  slug?: string;
  icon?: string;
  description?: string;
  order_index?: number;
  is_active?: boolean;
}

export interface TravelGuideCategoryUpdate {
  name?: string;
  slug?: string;
  icon?: string;
  description?: string;
  order_index?: number;
  is_active?: boolean;
}

export interface DestinationGuideItem {
  id: number;
  country_id: number;
  category_id: number;
  title: string;
  content: string;
  icon?: string;
  order_index: number;
  is_active: boolean;
  category?: TravelGuideCategory;
  created_at?: string;
  updated_at?: string;
}

export interface DestinationGuideItemCreate {
  country_id: number;
  category_id: number;
  title: string;
  content: string;
  icon?: string;
  order_index?: number;
  is_active?: boolean;
}

export interface DestinationGuideItemUpdate {
  category_id?: number;
  title?: string;
  content?: string;
  icon?: string;
  order_index?: number;
  is_active?: boolean;
}

File: backend/app/schemas/travel_guide.py
from typing import Optional, List
from datetime import datetime
from pydantic import BaseModel, Field

# Travel Guide Category Schemas
class TravelGuideCategoryBase(BaseModel):
    name: str = Field(..., max_length=100)
    slug: Optional[str] = Field(None, max_length=100)
    icon: Optional[str] = Field(None, max_length=50)
    description: Optional[str] = None
    order_index: int = 0
    is_active: bool = True

class TravelGuideCategoryCreate(TravelGuideCategoryBase):
    pass

class TravelGuideCategoryUpdate(BaseModel):
    name: Optional[str] = None
    slug: Optional[str] = None
    icon: Optional[str] = None
    description: Optional[str] = None
    order_index: Optional[int] = None
    is_active: Optional[bool] = None

class TravelGuideCategory(TravelGuideCategoryBase):
    id: int
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None

    class Config:
        from_attributes = True

# Travel Guide Item Schemas
class TravelGuideItemBase(BaseModel):
    country_id: int
    category_id: int
    title: str = Field(..., max_length=200)
    content: str
    icon: Optional[str] = Field(None, max_length=50)
    order_index: int = 0
    is_active: bool = True

class TravelGuideItemCreate(TravelGuideItemBase):
    pass

class TravelGuideItemUpdate(BaseModel):
    category_id: Optional[int] = None
    title: Optional[str] = None
    content: Optional[str] = None
    icon: Optional[str] = None
    order_index: Optional[int] = None
    is_active: Optional[bool] = None

class TravelGuideItem(TravelGuideItemBase):
    id: int
    category: Optional[TravelGuideCategory] = None
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None

    class Config:
        from_attributes = True
